### ISSUE TITLE
Doxygen main page missing

### ISSUE THREAD

**Issue by sekargovindarajan** [svenevs/exhale#100]
Hi, Thank you very much for this amazing tool.

Looks like Exhale is not picking up the contents of \mainpage from Doxygen. I do see the corresponding XML file. Should I do something special to make this happen?

My workaround is to create an mainpage.rst file and include it.
Thanks.

**Comment by svenevs** [comment]
Great question, haven't really accommodated all of the doxygen special commands.  Breathe handles most of that, but I do [some "miniparsing" for file page and namespace documentation](https://exhale.readthedocs.io/en/latest/usage.html#file-and-namespace-level-documentation-in-exhale) .  Looking at the [`\mainpage` docs](https://www.doxygen.nl/manual/commands.html#cmdmainpage) (emphasis from me on first sentence):

> If the `\mainpage` command is placed in a comment block the block is used to **customize the index page (in HTML)** or the first chapter (in LaTeX).
>
> The title argument is optional and replaces the default title that doxygen normally generates. If you do not want any title you can specify `notitle` as the argument of `\mainpage`.
>
> Here is an example:
>
> ```cpp
> /*! \mainpage My Personal Index Page
>  *
>  * \section intro_sec Introduction
>  *
>  * This is the introduction.
>  *
>  * \section install_sec Installation
>  *
>  * \subsection step1 Step 1: Opening the box
>  *
>  * etc...
>  */
> ```
>
> You can refer to the main page using: [`\ref`](https://www.doxygen.nl/manual/commands.html#cmdref) `index`.
>
> **See also**
>     section [\section](https://www.doxygen.nl/manual/commands.html#cmdsection), section [\subsection](https://www.doxygen.nl/manual/commands.html#cmdsubsection), and section [\page](https://www.doxygen.nl/manual/commands.html#cmdpage).

My read on this is that it's used to customize the `index.html` page, but maybe "index page" means something different here.  Cross-referencing the breathe issue you opened https://github.com/michaeljones/breathe/issues/619 but I think this issue makes more sense to solve with exhale, or possibly discourage usage of `\mainpage` depending on the discussion.

> My workaround is to create an `mainpage.rst` file and include it.

This definitely seems like a valid solution at least for now :upside_down_face:  But I guess the question is what is it supposed to do / what are you seeking?  Including extra information that you may already understand for any future readers, since the relationship between all of the tools is quite confusing :slightly_smiling_face:

1. The `index.rst` document that you include with sphinx is a "standalone" document, unrelated to doxygen, exhale, or breathe.  So, like creating a `mainpage.rst`, it's more or less entirely up to you.  When you use the [`.. toctree::` directive to link the generated files from exhale](), that directive is just linking to other pages but comes from sphinx.
2. It sounds like you seek to have this generated in the [generated root library document](https://exhale.readthedocs.io/en/latest/usage.html#controlling-the-layout-of-the-generated-root-library-document)?
3. Is your project open source / can you provide a small working example to help me understand the goal here?
4. Are you linking to it from other docs using `\ref index` anywhere?  This might be where breathe and exhale need to coordinate.

Basically, I'm open to fiddling with things / finding a good balance!  But when adopting sphinx, some things in doxygen you may need to abandon or alter and this may be one of them.  Either way, we should definitely document it at the very least!

**Comment by sekargovindarajan** [comment]
Thanks for the quick response (and sorry for the delay).

- After I posted it, I realized my fault with the breathe issue ([michaeljones/breathe#619](https://github.com/michaeljones/breathe/issues/619)), will close that.
- I also realize exhale is meant for code documentation that comes out of Doxygen, so I am *very, very, pleased* with it - so much more readable! Thanks again.
- My goal was to see if main page can be rendered as well, because usually people tend to put in very useful comments, design philosophy, etc. - although it is *not* code, it can convey the thinking of the author.
- Even without `\ref index`, I see that the XML file exists for main page.
- I do have an open source repo (not a project!), I will add the ability to create sphinx/breathe/exhale based doc, and post the link back here by the end of the day (US Eastern Time)

Again, if it's a lot of work, feel free to abandon it, especially since there's the `mainpage.rst` solution, and your tool is already doing amazing things!
